
# Genes Unknown in Acintetobacter Baumannii (GUNK)

![GUNK logo](https://www.patricbrc.org/public/patric/images/GUNK-logo.png)

**Program Director:** Samuel I. Miller MD, University of Washington

**Investigators:** Colin Manoil, Caroline S. Harwood, James E. Bruce, Mitchell J. Brittnacher, Hillary S. Hayden

**Project Home Page:** [http://uwgenomics.org/ab/index](http://uwgenomics.org/ab/index)

**Biological and Web Resources:** [http://uwgenomics.org/ab/tools](http://uwgenomics.org/ab/tools)

**Project Objectives:** Acinetobacter baumannii is an emerging pathogen that causes primary and opportunistic infections, and is a serious concern in the hospital and battlefield settings. This organism has significant intrinsic resistance to antibiotics as well as an ability to quickly acquire new resistance. Additionally, A. baumannii can persist in a desiccated state on fomites, making disinfection and management of nosocomial infections particularly troublesome. The goal of the GUNK center is to find unexploited therapeutic approaches by elucidating the role of uncharacterized genes of A. baumannii in this organism’s mechanisms of pathogenesis, resistance, and persistence.

## Publications

### Resources for Genetic and Genomic Analysis of Emerging Pathogen Acinetobacter baumannii

PMID:[25845845](https://www.ncbi.nlm.nih.gov/pubmed/?term=25845845)

Acinetobacter baumannii is a Gram-negative bacterial pathogen notorious for causing serious nosocomial infections that resist antibiotic therapy. Research to identify factors responsible for the pathogen’s success has been limited by the resources available for genome-scale experimental studies. This report describes the development of several such resources for A. baumannii strain AB5075, a recently characterized wound isolate that is multidrug resistant and displays robust virulence in animal models. We report the completion and annotation of the genome sequence, the construction of a comprehensive ordered transposon mutant library, the extension of high-coverage transposon mutant pool sequencing (Tn-seq) to the strain, and the identification of the genes essential for growth on nutrient-rich agar. These resources should facilitate large-scale genetic analysis of virulence, resistance, and other clinically relevant traits that make A. baumannii a formidable public health threat.

Data sets and additional information can be found [here](https://www.patricbrc.org/webpage/website/data_collections/content/gunk-experiment-1.html).

### Host-Microbe Protein Interactions during Bacterial Infection

PMID:[26548613](http://www.ncbi.nlm.nih.gov/pubmed/26548613)

Interspecies protein-protein interactions are essential mediators of infection. While bacterial proteins required for host cell invasion and infection can be identified through bacterial mutant library screens, information about host target proteins and interspecies complex structures has been more difficult to acquire. Using an unbiased chemical crosslinking/mass spectrometry approach, we identified interspecies protein-protein interactions in human lung epithelial cells infected with Acinetobacter baumannii. These efforts resulted in identification of 3,076 crosslinked peptide pairs and 46 interspecies protein-protein interactions. Most notably, the key A. baumannii virulence factor, OmpA, was identified as crosslinked to host proteins involved in desmosomes, specialized structures that mediate host cell-to-cell adhesion. Co-immunoprecipitation and transposon mutant experiments were used to verify these interactions and demonstrate relevance for host cell invasion and acute murine lung infection. These results shed new light on A. baumannii-host protein interactions and their structural features, and the presented approach is generally applicable to other systems.

Data sets and additional information can be found [here](https://www.patricbrc.org/webpage/website/data_collections/content/gunk-experiment-2.html).

### Joint Transcriptional Control of Virulence and Resistance to Antibiotic and Environmental Stress in Acinetobacter baumannii

PMID:[26556274](https://www.ncbi.nlm.nih.gov/pubmed/?term=26556274)

The increasing emergence of antibiotic-resistant bacterial pathogens represents a serious risk to human health and the entire health care system. Many currently circulating strains of Acinetobacter baumannii exhibit resistance to multiple antibiotics. A key limitation in combating A. baumannii is that our understanding of the molecular mechanisms underlying the pathogenesis of A. baumannii is lacking. To identify potential virulence determinants of a contemporary multidrug-resistant isolate of A. baumannii, we used transposon insertion sequencing (TnSeq) of strain AB5075. A collection of 250,000 A. baumannii transposon mutants was analyzed for growth within Galleria mellonella larvae, an insect-based infection model. The screen identified 300 genes that were specifically required for survival and/or growth of A. baumannii inside G. mellonella larvae. These genes encompass both known, established virulence factors and several novel genes. Among these were more than 30 transcription factors required for growth in G. mellonella. A subset of the transcription factors was also found to be required for resistance to antibiotics and environmental stress. This work thus establishes a novel connection between virulence and resistance to both antibiotics and environmental stress in A. baumannii.

Acinetobacter baumannii is rapidly emerging as a significant human pathogen, largely because of disinfectant and antibiotic resistance, causing lethal infection in fragile hosts. Despite the increasing prevalence of infections with multidrug-resistant A. baumannii strains, little is known regarding not only the molecular mechanisms that allow A. baumannii to resist environmental stresses (i.e., antibiotics and disinfectants) but also how these pathogens survive within an infected host to cause disease. We employed a large-scale genetic screen to identify genes required for A. baumannii to survive and grow in an insect disease model. While we identified many known virulence factors harbored by A. baumannii, we also discovered many novel genes that likely play key roles in A. baumannii survival of exposure to antibiotics and other stress-inducing chemicals. These results suggest that selection for increased resistance to antibiotics and environmental stress may inadvertently select for increased virulence in A. baumannii.

Genes required for growth of A. baumannii in G. mellonella are available [here](http://brcdownloads.patricbrc.org/BRC_Mirrors/GUNK/pmid_26556274/pmid_26556274_table_1.xlsx).

### In vivo protein interaction network analysis reveals porin-localized antibiotic inactivation in Acinetobacter baumannii strain AB5075

PMID:[27834373](https://www.ncbi.nlm.nih.gov/pubmed/27834373)

The nosocomial pathogen Acinetobacter baumannii is a frequent cause of hospital-acquired infections worldwide and is a challenge for treatment due to its evolved resistance to antibiotics, including carbapenems. Here, to gain insight on A. baumannii antibiotic resistance mechanisms, we analyse the protein interaction network of a multidrug-resistant A. baumannii clinical strain (AB5075). Using in vivo chemical cross-linking and mass spectrometry, we identify 2,068 non-redundant cross-linked peptide pairs containing 245 intra- and 398 inter-molecular interactions. Outer membrane proteins OmpA and YiaD, and carbapenemase Oxa-23 are hubs of the identified interaction network. Eighteen novel interactors of Oxa-23 are identified. Interactions of Oxa-23 with outer membrane porins OmpA and CarO are verified with co-immunoprecipitation analysis. Furthermore, transposon mutagenesis of oxa-23 or interactors of Oxa-23 demonstrates changes in meropenem or imipenem sensitivity in strain AB5075. These results provide a view of porin-localized antibiotic inactivation and increase understanding of bacterial antibiotic resistance mechanisms.

Cross-linked peptide pairs and protein-protein interactions identified in this study are available [here](http://brcdownloads.patricbrc.org/BRC_Mirrors/GUNK/pmid_27834373/).

### Importance of Core Genome Functions for an Extreme Antibiotic Resistance Trait

PMID:[29233894](https://www.ncbi.nlm.nih.gov/pubmed/29233894)

Extreme antibiotic resistance in bacteria is associated with the expression of powerful inactivating enzymes and other functions encoded in accessory genomic elements. The contribution of core genome processes to high-level resistance in such bacteria has been unclear. In the work reported here, we evaluated the relative importance of core and accessory functions for high-level resistance to the aminoglycoside tobramycin in the nosocomial pathogen Acinetobacter baumannii Three lines of evidence establish the primacy of core functions in this resistance. First, in a genome scale mutant analysis using transposon sequencing and validation with 594 individual mutants, nearly all mutations reducing tobramycin resistance inactivated core genes, some with stronger phenotypes than those caused by the elimination of aminoglycoside-inactivating enzymes. Second, the core functions mediating resistance were nearly identical in the wild type and a deletion mutant lacking a genome resistance island that encodes the inactivating enzymes. Thus, most or all of the core resistance determinants important in the absence of the enzymes are also important in their presence. Third, reductions in tobramycin resistance caused by different core mutations were additive, and highly sensitive double and triple mutants (with 250-fold reductions in the MIC) that retained accessory resistance genes could be constructed. Core processes that contribute most strongly to intrinsic tobramycin resistance include phospholipid biosynthesis, phosphate regulation, and envelope homeostasis.

The inexorable increase in bacterial antibiotic resistance threatens to undermine many of the procedures that transformed medicine in the last century. One strategy to meet the challenge antibiotic resistance poses is the development of drugs that undermine resistance. To identify potential targets for such adjuvants, we identified the functions underlying resistance to an important class of antibiotics for one of the most highly resistant pathogens known.

Candidate tobramycin resistance genes identified in AB5075 and AB5075ΔRI are available [here](http://brcdownloads.patricbrc.org/BRC_Mirrors/GUNK/pmid_29233894/).
