TB-ARC CDRC
===========

.. image:: images/broad.jpg

`TB-ARC CDRC at Broad <https://olive.broadinstitute.org/projects/tb_cdrc>`_

The NIH-funded Tuberculosis Clinical Diagnostics Research Consortium (TB-CDRC) (Contract No. HHSN272200900050C), provided approximately 200 clinical Mycobacterium tuberculosis strains isolated from humans with TB disease, with a wide variety of drug resistance patterns from Uganda and South Korea, for whole genome sequencing. The strains provide a unique resource because they have had extensive characterization of their drug susceptibility profiles, including minimum inhibitory concentration (MIC) data for most first- and second-line drugs as well as breakpoint resistance data established in both liquid and solid media. The sequenced strains have also been deposited into BEI Resources for use by the scientific community. This project is highly relevant to the understanding of the biology of M. tuberculosis and to development of new diagnostic and therapeutic approaches.

Solid culture based phenotypic testing remains the stalwart gold standard for antimicrobial susceptibility testing. However, genotypic diagnostic tools have the potential to standardize and speed up the drug susceptibility testing process, while also reducing biohazard issues. Specifically, identification of the complement of mutations which do (and do not) encode for drug resistance will enable the community to develop diagnostic tests for detection of drug resistance that cover a broader spectrum of mutations and have better sensitivity and specificity compared to those currently available. In addition, the deposited, well characterized strain bank will provide a novel diagnostic testing panel for analyzing the accuracy of new tests and allowing for easy head-to-head comparison of tests. This will speed up the test approval process and reduce the need for expensive early field-testing.

With respect to development of new anti-TB therapeutics, the production of an annotated SNP map linked with different drug resistance MICs will help establish novel resistance mechanisms and will facilitate studies to better understand the mechanisms of drug resistance evolution. This resource will enable the scientific community to develop novel therapeutic approaches to prevent the emergence of resistance in TB.

With respect to drug resistance profiles, the strains are heterogeneous, but within the set of 200 there are at least 50 sensitive and at least 50 resistant strains for each individual drug.

With respect to M. tuberculosis phylogenetics, epidemiological data from the Republic of Korea have shown that the majority of strains are K family strains, derivatives of the Beijing family (Kang et al., Journal of Medical Microbiology 59, 1191-1197). In the Kampala, Uganda vicinity (where the Ugandan strains were collected), the T2 spoligotype family of strains has been shown to predominate. Therefore we anticipate that the phylogenetic distribution within the proposed strain set is reasonably restricted, which should facilitate sequencing data interpretation. Yet, the two strain sets (Ugandan versus Korean) are different from each other, suggesting that the data provided will be representative of the broader M. tuberculosis population.

Data download:

- `Broad original files <ftp://ftp.patricbrc.org/BRC_Mirrors/TB-ARC/broad_original/CDRC.1/variants.tar.gz>`_
- `PATRIC mapped files <ftp://ftp.patricbrc.org/BRC_Mirrors/TB-ARC/patric_mapped/CDRC.1.tar.gz>`_
- `PATRIC FTP site <http://brcdownloads.patricbrc.org/BRC_Mirrors/TB-ARC/patric_mapped/CDRC.1/>`_

View data in PATRIC:

- Download the PATRIC mapped files (above)
- Go to: `<https://www.patricbrc.org/view/Genome/83332.111#view_tab=browser>`_
- Click on File -> Open -> Select Files and upload both the corresponding .vcf.gz and .vcf.gz.tbi files, keep the default options for track type, and click Open button
