TB-ARC K-RITH KwaZulu-Natal
===========================

.. image:: images/broad.jpg

`TB-ARC K-RITH KwaZulu-Natal at Broad <https://olive.broadinstitute.org/projects/tb_krith_kzn>`_

The emergence of drug-resistant tuberculosis (TB) has complicated global TB control efforts. The World Health Organization (WHO) estimates that up to half a million multidrug-resistant TB (MDR-TB) cases occur annually.

Diagnostic limitations are a major barrier to the recognition and subsequent management of drug-resistant TB. Like other infectious organisms, Mycobacterium tuberculosis (Mtb) is known to develop drug resistance via acquisition of mutations in its DNA. These genomic mutations often arise during the course of drug therapy as the result of non-compliance or inappropriate drug regimens, which create selective drug pressure that leads to the development of resistant strains.

For each anti-TB drug, there are several known Mtb genetic mutations commonly associated with bacterial drug resistance. However, these genetic mutations do not account for all known drug resistance to their cognate anti-TB drugs. Furthermore, it is likely that there are yet unidentified Mtb genes and molecular mechanisms that are associated with drug resistance.

The overarching goal for this work is to obtain full genome sequence data from well characterized drug-resistant, MDR-TB, XDR-TB (extensively drug resistant TB), and TDR-TB (totally drug resistant TB) strains to understand the genomic patterns of drug resistance in support of:

1. Predictive analytics platforms and services to support patient treatment
2. Development of new diagnostics
3. New drug discovery
4. Molecular epidemiology for population control programs including drug supply chain management

Progress in any one of these four areas will have an immediate and positive impact on the global TB burden by providing clinicians with real-time information and better treatment options for new cases of TB.

We have sequenced over 200 Mtb strains from two patient cohorts from the province of KwaZulu-Natal (KZN) in South Africa. Each isolate has been characterized by clinical laboratory DST, confirmed by at least one other laboratory, and is accompanied by other required minimum metadata. Sequence polymorphisms will be identified and compared across genomes derived from both cohorts to identify the suite of polymorphisms corresponding to each drug resistance profile. This information will inform future work to sequence Mtb from patient cohorts in this and other locales.

Deeper understanding of the genomic basis of resistance will allow for much-needed improvements in the design of diagnostics of drug-resistance. Beyond understanding and predicting resistance patterns, whole genome sequencing will provide an avenue to investigate much broader basic science and epidemiological research questions with respect to TB. Molecular epidemiology will aid definition of TB strain determination, distribution and spread of infection across a geographic region and evolution of epidemics over time.

Data download:

- `Broad original files <ftp://ftp.patricbrc.org/BRC_Mirrors/TB-ARC/broad_original/KwaZulu-Natal.3/TB-ARC_K-RITH1_SNPdata.tar.gz>`_
- `PATRIC mapped files <ftp://ftp.patricbrc.org/BRC_Mirrors/TB-ARC/patric_mapped/KwaZulu-Natal.3.tar.gz>`_
- `PATRIC FTP site <http://brcdownloads.patricbrc.org/BRC_Mirrors/TB-ARC/patric_mapped/KwaZulu-Natal.3/>`_

View data in PATRIC:

- Download the PATRIC mapped files (above)
- Go to: `<>`_
- Click on File -> Open -> Select Files and upload both the corresponding .vcf.gz and .vcf.gz.tbi files, keep the default options for track type, and click Open button