==========================================
PATRIC Driving Biological Projects Awarded
==========================================


:Date:   2010-10-29T20:34:09+00:00

The Pathosystems Resource Integration Center
(`PATRIC <http://www.patricbrc.org/>`__), a bacterial Bioinformatics
Resource Center (BRC) funded by the National Institute of Allergy and
Infectious Diseases (NIAID) of the National Institutes of Health (NIH),
is pleased to announce awards for its Driving Biological Projects
(DBPs). DBPs are two-year awards that focus on infectious diseases
research related to human bacterial pathogens. Research includes the use
of high-throughput experimental technologies (HTP) to functionally
characterize the genome, proteome or metabolome of bacterial organisms
and/or host-pathogen interactions. This research may help elucidate the
role of genes, proteins and metabolites with respect to pathogenesis,
antimicrobial resistance or other biological processes of interest. The
data and information generated by the DBPs will be made accessible to
the broad scientific community and will be used to drive PATRIC’s
infrastructure development to enable other researchers to perform
similar analyses at the PATRIC website using their own data.

The DBP proposals were independently reviewed and evaluated by the
PATRIC Scientific Working Group
(`SWG <http://patricbrc.org/portal/portal/patric/About?page=swg>`__),
and the top two proposals were selected for award.  The awarded DBPs
are:

-  Comparative transcriptome and proteome analysis of Clostridium
   difficile strains: Dr. Yung-Fu Chang, PI, Dept. of Population
   Medicine and Diagnostic Sciences, College of Veterinary Medicine,
   Cornell University – The Cornell team will compare five *C.
   difficile* strains, generate transcriptome and proteome sequence data
   for these strains, verify and complement the genome sequence
   annotations of available sequenced strains and expand the PATRIC data
   model by joint development, testing and deployment of tools for
   RNA-Seq analysis that can be used for other bacterial RNA-Seq
   projects.
-  Fitness Annotation of Bacterial Genomes: Dr. Michael McClelland, PI,
   Dept. of Pathology and Laboratory Medicine, University of California,
   Irvine – The goal of this proposal is to provide PATRIC with the
   information needed to display which genes of non-typhoidal
   *Salmonella* serovar Typhimurium (STm) contribute to survival in a
   variety of environments, including in infections of various hosts.
   This will be accomplished through a combination of high-throughput
   screening and sequencing methods and unique resources developed to
   annotate the STm genome with fitness information.   STm
   transcriptomes will be generated from bacteria growing in defined
   environments, including rich and minimal media, at stationary phase,
   and in conditions that induce virulence pathways. These
   transcriptomes will provide basal reference profiles to standardize
   and improve analysis of the impending onslaught of high-throughput
   transcriptomics data. Decoration of the genome with basal
   transcriptional data will complement the fitness profile and other
   existing annotations in this archetypical strain.

PATRIC will use the information learned and workflows identified in
these DBPs to develop Web-accessible infrastructure to accept and
analyze RNA-Seq data and other associated data types.  The knowledge
gained through the experiments will be used to improve annotations for
the specific organisms under study and in a broader context where
applicable.

Dr. Chang said, “The PATRIC DBP is an excellent opportunity for our
laboratory to expand the ongoing genomics research on *C. difficile*. In
this DBP, we will be integrating large scale RNA-Seq, proteome and
phenotype array data into the genome annotations of *C. difficile*. The
experimental data generated will further be used for extending the
annotations to other draft *C. difficile* genomes. We will develop
experimentally curated pathway maps for several C. difficile genomes. We
will also develop a pathway cross-mapping system between popular pathway
analysis tools such as KEGG, SEED and Pathway Tools. Dr. Joy Scaria from
the Chang laboratory will also work with the PATRIC team at VBI to
seamlessly integrate the experimental data into PATRIC system. With the
active participation of PATRIC team in the data analysis and software
deployment, we anticipate to develop a scalable genomic analysis model
extendable to other bacterial pathogens.”

Dr. McClelland said, “It is a privilege to be part of this vitally
important NIAID mechanism in which wet labs are funded to link up with
bioinformatics resource groups and make their data and analysis tools
quickly and easily available to the community. There is a desperate need
for many more of these kinds of grants to be funded to keep up with the
increasing power of high-throughput data generation and presentation
strategies, and to include a wide a range of pathogens and experts.”

Dr. Bruno Sobral, PI of the PATRIC project, offered the following
comments on the DBPs: “PATRIC fully supports the concept of driving
biological projects with the infectious disease community that performs
genomic-scale wet chemistry experiments. The excellent quality of all
the proposals received and especially the two selected shows that there
is much to be done in support of the thousands of laboratories that work
on bacterial infectious diseases. We are grateful to expand the PATRIC
team to include such excellent scientists working on two organisms of
great importance for hospital-acquired infections and food-borne
outbreaks.”

Another solicitation for PATRIC DBPs is expected to occur in 2012.  DBP
announcements will be posted on the PATRIC Website.  For individuals who
wish to be included on the PATRIC email list for updates to the site,
workshop notifications and DBP announcements, please contact us at
patric@vbi.vt.edu

**About PATRIC**

`PATRIC <http://www.patricbrc.org/>`__, the PathoSystems Resource
Integration Center is a comprehensive Web-based resource for bacterial
pathogens, biodefense research, and the study of emerging infectious
diseases. PATRIC acquires, stores and integrates diverse information
from pathogenic bacteria, provides visualization and data analysis tools
to scientists, and enables the analysis of genomic, proteomic and
various other data arising from, and of relevance to, infectious disease
research. PATRIC’s scope includes all bacterial species and near
relatives in the selected category A-C pathogens list as defined by
NIAID.

The PATRIC project includes three primary collaborators: the University
of Chicago, the University of Manchester, and New City Media. The
University of Chicago is providing genome annotations and a PATRIC
end-user genome annotation service using their Rapid Annotation using
Subsystem Technology (RAST) system. The National Centre for Text Mining
(NaCTeM) at the University of Manchester is providing literature-based
text mining capability and service. New City Media is providing
assistance in website interface development. The PATRIC project also
develops and hosts the `Pathogen Portal <http://pathogenportal.org/>`__
a gateway and common data repository for the entire NIAID BRC Program.

This project has been funded in whole or in part with Federal funds from
the National Institute of Allergy and Infectious Diseases, National
Institutes of Health, Department of Health and Human Services, under
Contract No. HHSN272200900040C.

**About the Cyberinfrastructure Division**

The Cyberinfrastructure (CI) Division at the Virginia Bioinformatics
Institute (VBI) of Virginia Tech (VT) is a diverse and highly integrated
team of biologists, bioinformaticians, software engineers, and other
experts that develops data, methods, infrastructure, and resources that
enable scientific discoveries in infectious disease research and
increasingly other research fields. The team applies the principles of
cyberinfrastructure to integrate data, computational infrastructure, and
people. The CI Division has developed many public resources for curated,
diverse molecular and literature data from various systems, and
implemented the processes, systems, and databases required to support
them. Members of the team also conduct research their own research in a
highly coordinated way within the Division and with scientific
communities across the globe by applying its methods, data, and
infrastructure to make new discoveries.
