:github_url: https://github.com/PATRIC3/p3_docs/blob/master/docroot/news/20130326-bacterial-dna-sifted-from-the-trichoplax-adhaerens-animaliaplacozoa-genome-project-reveals-a-putative-rickettsial-endosymbiont.rst

=================================================================================================================================
Bacterial DNA sifted from the Trichoplax adhaerens (Animalia:Placozoa) genome project reveals a putative rickettsial endosymbiont
=================================================================================================================================


:date:   2013-03-26T12:41:20+00:00

 

Driscoll, T., J.J. Gillespie, E.K. Nordberg, A.F Azad, and B.W. Sobral
(2013). **Bacterial DNA sifted from the *Trichoplax adhaerens*
(Animalia:Placozoa) genome project reveals a putative rickettsial
endosymbiont.** Genome Biol Evol 5(4): 621-645.  doi:
10.1093/gbe/evt036. PMID:  23475938

Click here for supplemental information and additional data.

**Abstract**

Eukaryotic genome sequencing projects often yield bacterial DNA
sequences, data typically considered as microbial contamination.
However, these sequences may also indicate either symbiont genes or
lateral gene transfer (LGT) to host genomes. These bacterial sequences
can provide clues about eukaryote-microbe interactions. Here, we used
the genome of the primitive animal Trichoplax adhaerens (Metazoa:
Placozoa), which is known to harbor an uncharacterized Gram-negative
endosymbiont, to search for the presence of bacterial DNA sequences.
Bioinformatic and phylogenomic analyses of extracted data from the
genome assembly (181 bacterial CDS) and trace read archive (16S rDNA)
revealed a dominant proteobacterial profile strongly skewed to
Rickettsiales (Alphaproteobacteria) genomes. By way of phylogenetic
analysis of 16S rDNA and 113 proteins conserved across proteobacterial
genomes, as well as identification of 27 rickettsial signature genes, we
propose a Rickettsiales endosymbiont of Trichoplax adhaerens (RETA). The
majority (93%) of the identified bacterial CDS belong to small scaffolds
containing prokaryotic-like genes; however, 12 CDS were identified on
large scaffolds comprised of eukaryotic-like genes, suggesting that T.
adhaerens might have recently acquired bacterial genes. These putative
LGTs may coincide with the placozoan‚s aquatic niche and symbiosis with
RETA. This work underscores the rich, and relatively untapped, resource
of eukaryotic genome projects for harboring data pertinent to
hostmicrobial interactions. The nature of unknown (or poorly
characterized) bacterial species may only emerge via analysis of host
genome sequencing projects, particularly if these species are resistant
to cell culturing, as are many obligate intracellular microbes. Our work
provides methodological insight for such an approach.
