=================================================================
Standardized metadata for human pathogen/vector genomic sequences
=================================================================


:date:   2014-06-20T08:08:00+00:00

Dugan V.G., S.J. Emrich, G.I. Giraldo-Calderón, O.S. Harb, R.M. Newman,
B.E. Pickett, L.M. Schriml, T.B. Stockwell, C. J. Stoeckert Jr, D. E.
Sullivan, I. Singh, D.V. Ward, A. Yao, J. Zheng, T. Barrett, B. Birren,
L. Brinkac,V.M. Bruno, E. Caler, S. Chapman, F.H. Collins, C.A. Cuomo,
V. Di Francesco, S. Durkin, M. Eppinger, M. Feldgarden, C. Fraser, W.F.
Fricke, M. Giovanni, M.R. Henn, E. Hine, J.D. Hotopp, I.
Karsch-Mizrachi, J.C. Kissinger, E.M. Lee, P. Mathur, E.F. Mongodin,
C.I. Murphy, G. Myers, D.E. Neafsey, K.E. Nelson, W.C. Nierman, J.
Puzak, D. Rasko, D.S. Roos, L. Sadzewicz, J.C. Silva, B. Sobral, R.B.
Squires, R.L. Stevens, L. Tallon, H. Tettelin, D. Wentworth, O. White,
R. Will, J. Wortman, Y. Zhang, R.H. Scheuermann. (2014). “**Standardized
metadata for human pathogen/vector genomic sequences**.” PLoS ONE 9(6):
e99979. doi: 10.1371/journal.pone.0099979. PMID:  24936976.

**Abstract**

High throughput sequencing has accelerated the determination of genome
sequences for thousands of human infectious disease pathogens and dozens
of their vectors. The scale and scope of these data are enabling
genotype-phenotype association studies to identify genetic determinants
of pathogen virulence and drug/insecticide resistance, and phylogenetic
studies to track the origin and spread of disease outbreaks. To maximize
the utility of genomic sequences for these purposes, it is essential
that metadata about the pathogen/vector isolate characteristics be
collected and made available in organized, clear, and consistent
formats. Here we report the development of the GSCID/BRC Project and
Sample Application Standard, developed by representatives of the Genome
Sequencing Centers for Infectious Diseases (GSCIDs), the Bioinformatics
Resource Centers (BRCs) for Infectious Diseases, and the U.S. National
Institute of Allergy and Infectious Diseases (NIAID), part of the
National Institutes of Health (NIH), informed by interactions with
numerous collaborating scientists. It includes mapping to terms from
other data standards initiatives, including the Genomic Standards
Consortium’s minimal information (MIxS) and NCBI’s BioSample/BioProjects
checklists and the Ontology for Biomedical Investigations (OBI). The
standard includes data fields about characteristics of the organism or
environmental source of the specimen, spatial-temporal information about
the specimen isolation event, phenotypic characteristics of the
pathogen/vector isolated, and project leadership and support. By
modeling metadata fields into an ontology-based semantic framework and
reusing existing ontologies and minimum information checklists, the
application standard can be extended to support additional
project-specific data fields and integrated with other data represented
with comparable standards. The use of this metadata standard by all
ongoing and future GSCID sequencing projects will provide a consistent
representation of these data in the BRC resources and other repositories
that leverage these data, allowing investigators to identify relevant
genomic sequences and perform comparative genomics analyses that are
both statistically meaningful and biologically relevant.
