BV-BRC Viral Sub-species Classification Workshop at NIAID Rockville, MD, April 8-10, 2024
=========================================================================================

.. raw:: html

- `Introduction <#introduction-section>`_
- `Workshop Highlights <#highlights-section>`_
- `Workshop Agenda and Presentations <#agenda-section>`_
- `Workshop Speakers and Reading List <#speakers-section>`_

.. _introduction-section:

------------------
Introduction
------------------

---------------------

A critical aspect of the response to the COVID-19 pandemic was to trace, classify, and understand the evolving virus lineages, their phenotypic differences, and their impact on diagnostics, vaccines, and therapeutics. Classification and tracing of sub-species level variation has also been similarly important in outbreaks of viruses such as influenza, HIV, and monkeypox, and has informed public health risk assessment and communication strategies. To prepare for the next, inevitable outbreak of virus disease, the Bacterial and Viral Bioinformatics Resource Center (BV-BRC) along with the CDC, NCBI, and NIAID, HOSTED a hybrid workshop on viral sub-species classification. Workshop objectives include the review of various classification schemes developed for SARS-CoV-2 and other viruses, examination of the methods and tools developed for classification, discussion of the utility and impact of classification schemes on the public health response, and review of lessons learned from past and current outbreaks. The outcome of the workshop will be to define best practices for establishing sub-species classification systems for these and other viruses that will be important in controlling future out breaks and potential pandemics. The valuable insights and best practices from this workshop will be widely disseminated.

.. _highlights-section:

------------------
Workshop Highlights
------------------

--------------------------

Session One, led by Elliot Lefkowitz, tracked virus evolution and its impact on sub-species. Dr. Richard Neher discussed RNA virus spread, followed by Dr. Bette Korber's insights on SARS-CoV-2 and HIV variants.

Session Two, moderated by Dr. Rodney Brister, explored RNA and DNA virus complexity. Dr. Justin Bahl discussed RNA virus evolution, Dr. Tavis Anderson spoke on influenza A virus, and Dr. Jens Kuhn covered Filovirus nomenclature.

Session Three, moderated by Dr. Conrad Schoch, explored diverse perspectives on virus classification, featuring discussions on Influenza, SARS-CoV-2, and virus classification from Dr. Derek Smith and Dr. Peter Simmonds, respectively.

Session Four, moderated by Dr. Richard Scheuermann, delved into classification schemes, featuring discussions from Dr. Jens H. Kuhn on ICTV taxonomy, Dr. Krista Queen about GISAID classification, Dr. AÃÅine O'Toole, on the Pango system development, and its future beyond the SARS-CoV-2 pandemic from Dr. Rachel Colquhoun.

Session Five, moderated by Dr. Eneida Hatcher, focused on tools for viral evolution classification. Dr. Cornelius Roemer discussed Clade and lineage assignment, Dr. Angie Hinrichs presented UShER and autolin, and Dr. Christian Zmasek discussed BV-BRC sub-species classification tools.

Session Six, moderated by Indresh Singh, explored analytical and visualization tools for studying virus evolution during outbreaks. Dr. Eneida Hatcher discussed NCBI Virus and Taxonomy, Dr. Philippe Le Mercier presented Virus sub-species in ViralZone and UniProtKB, and Dr. Andrew Warren discussed virus mutational spectra visualization.

Session Seven, led by Dr. Paul Eder, addressed public health responses. Dr. Brown discussed NIAID's disease prevention efforts, Dr. Beck shared CDC's perspective, Dr. Rosamund Lewis explored viral evolution's impact on pox outbreaks, and Dr. Xiong (Sean) Wang provided insights from the Minnesota Department of Health on virus surveillance.


.. _agenda-section:

---------------
Workshop Agenda with Links to Recordings and Presentations
---------------

---------------------

.. raw:: html

    </style>

    <p style="margin-bottom:3.0pt;">Agenda, Questions and Discussions at Reddit <a href="https://www.reddit.com/r/BRC_users/" target="_blank">here</a></strong></p>
    <p style="margin-bottom:3.0pt;"><strong>April 8, Day 1: Virus Sub-species Classification: What and Why</strong></p>
    <table class="docutils align-default workshop-agenda">
        <colgroup>
            <col style="width: 23.0%">
            <col style="width: 17.0%">
            <col style="width: 30.0%">
            <col style="width: 30.0%">
        </colgroup>
        <thead>
            <tr>
                <th>Time</th>
                <th>Topic</th>
                <th>Title</th>
                <th>Presenter</th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td>9:30 AM - 10:00 AM</td>
                <td>Welcome and Introduction</td>
                <td>Goals, Expectations, and Logistics</td>
                <td><ul class="no-bullets"><li style="margin-left: 0px;">Elliot Lefkowitz (UAB)</li><li style="margin-left: 0px; line-height: 1.2;">Wiriya Rutvisuttinunt (NIAID) on behalf of the Organizing Committee</li></ul></td>
            </tr>
            <tr>
                <td>10:00 AM - 12:00 PM</td>
                <td colspan="3">
                    <p><strong>Session I: Tracking virus evolution through sub-species classification</strong></p>
                    <p style="margin:3.0pt 0;"><em>The evolution of viruses during disease outbreaks and the impact this process has on sub-species classification, especially considering rapid rates of virus evolution.</em></p>
                    <br>
                    <p><strong>Moderator</strong></p>
                    <ul class="no-bullets"><li>Elliot Lefkowitz, PhD, University of Alabama at Birmingham</li></ul>
                    <br>
                    <p><strong>Panel and Session Topics</strong> [30-minute talks]</p>
                    <ul class="no-bullets">
                        <li><strong><em>Tracking and predicting the spread and evolution of RNA viruses</em></strong></li>
                        <li>Richard Neher, PhD, University of Basel, Switzerland</li>
                        <li>&nbsp;</li>
                        <li><strong><em>SARS-CoV-2 and HIV variant evolution</em></strong></li>
                        <li>Bette Korber, PhD, Los Alamos National Labs, U.S.</li>
                        <li>Recording <a href="https://www.reddit.com/r/BRC_users/comments/1bu52gn/feedback_requested_session_i_tracking_virus" target="_blank">here</a>., Presentation Download <a href="https://www.reddit.com/r/BRC_users/comments/1bu52gn/feedback_requested_session_i_tracking_virus" target="_blank">here</a>.</li>


                    </ul>
                    <br>
                    <p><strong>Break</strong> [15 minutes]</p>
                    <br>
                    <p><strong>Q&A Panel Discussions</strong> [45 Minutes]</p>
                    <ul>
                        <li>How are viruses classified below the level of species?</li>
                        <li>How does virus evolution impact outbreak response?</li>
                        <li>How have past outbreaks challenged traditional/current virus classification?</li>
                        <li>What are the implications of these findings for future outbreak preparedness and response?</li>
                    </ul>
                    <br>
                    <p>Use Reddit for questions and discussions <a href="https://www.reddit.com/r/BRC_users/comments/1bu52gn/feedback_requested_session_i_tracking_virus" target="_blank">here</a>.
                </td>
            </tr>
            <tr>
                <td>12:00 PM - 1:00 PM</td>
                <td colspan="3">Lunch</td>
            </tr>
            <tr>
                <td>1:00 PM - 3:05 PM</td>
                <td colspan="3">
                    <p><strong>Session II: Unraveling the complexities of RNA and DNA viruses</strong></p>
                    <p style="margin:3.0pt 0;"><em>Examples of the evolution of virus disease, the impact of lineage evolution on pathogenicity, and the consequences for classifying and responding to particular virus threats.</em></p>
                    <br>
                    <p><strong>Moderator</strong></p>
                    <ul class="no-bullets"><li>James Rodney Brister, PhD, NCBI</li></ul>
                    <br>
                    <p><strong>Panel and Session Topics</strong> [15-minute talks]</p>
                    <ul class="no-bullets">
                        <li><strong><em>Evolution, epidemiology, and RNA virus disease</em></strong></li>
                        <li>Justin Bahl, PhD, University of Georgia</li>
                        <li>&nbsp;</li>
                        <li><strong><em>The evolution of influenza A virus at the human-animal interface</em></strong></li>
                        <li>Tavis Anderson, PhD, U.S. Department of Agriculture</li>
                        <li>&nbsp;</li>
                        <li><strong><em>Filovirus sub-species nomenclature</em></strong></li>
                        <li>Jens H. Kuhn, PhD, NIH/NIAID/DCR Integrated Research Facility at Fort Detrick</li>
                        <li>&nbsp;</p>
                        <li><strong><em>Rotavirus diversity, evolution, and lineage classification</em></strong></li>
                        <li>Celeste Donato, PhD, Murdoch Children's Research Institute</li>
                    </ul>
                    <br>
                    <p><strong>Break</strong></p>
                    <br>
                    <ul class="no-bullets">
                        <li><strong><em>Update on Human Adenoviruses</em></strong></li>
                        <li>Donald Seto, PhD, George Mason University</p>
                        <li>&nbsp;</p>
                        <li><strong><em>Monkeypox virus sub-species evolution</em></strong></li>
                        <li>Elliot Lefkowitz, PhD, University of Alabama at Birmingham</li>
                    </ul>
                    <br>
                    <p><strong>Q&A Panel Discussions</strong> [20 Minutes]</p>
                    <ul>
                        <li>How are viruses classified?</li>
                        <li style="list-style-type: none !important;">
                            <ul>
                                <li>What biological and genetic features are used to support classification?</li>
                                <li>Are complete genomic sequences required to support classification or are specific subgenomic regions sufficient?</li>
                                <li>Are there common characteristics that can be used to classify all viruses or do distinguishing characteristics require multiple classification schemes?</li>
                            </ul>
                        </li>
                        <li>Is the current sub-species classification system for each virus sufficient to capture the relevant genetic diversity now and in the future?</li>
                        <li>During disease outbreaks, are there specific mutations or common patterns of genetic variation observed in RNA or DNA viruses that impact pathogenesis?</li>
                    </ul>
                    <br>
                    <p>Use Reddit for questions and discussions <a href="https://www.reddit.com/r/BRC_users/comments/1bu54kq/feedback_requested_session_ii_unraveling_the" target="_blank">here</a>.
                </td>
            </tr>
            <tr>
                <td>3:05 PM - 3:45 PM</td>
                <td colspan="3">Break <em>Darkness descends on us all; big and small. Alive? Not necessarily all.</em></td>
            </tr>
            <tr>
                <td>3:45 PM - 5:00 PM</td>
                <td colspan="3">
                    <p><strong>White Board Brainstorming Session</strong></p>
                    <br>
                    <p><strong>Moderators:</strong></p>
                    <ul class="no-bullets">
                        <li>Elodie Ghedin, PhD, NIAID</p>
                        <li>Session Moderators</li>
                    </ul>
                    <br>
                    <p><strong>Questions:</strong></p>
                    <ul>
                        <li>To what extent can we leverage our understanding of virus evolution to improve classification, predict novel, impactful strains, develop more effective diagnostics and vaccines, and ultimately inform control and response strategies?</li>
                    </ul>
                </td>
            </tr>
        </tbody>
    </table>

    <p style="margin-bottom:3.0pt;"><strong>April 9, Day 2: Virus Sub-species Classification: How</strong></p>
    <table class="docutils align-default workshop-agenda">
        <colgroup>
            <col style="width: 23.0%">
            <col style="width: 77.0%">
        </colgroup>
        <tbody>
            <tr>
                <td>9:30 AM - 9:35 AM</td>
                <td><strong>Welcome back and Recap of Day 1</strong></td>
            </tr>
            <tr>
                <td>9:35 AM - 10:45 AM</td>
                <td>
                    <p><strong>Session III: Diverse Viewpoints: Exploring Classification from Different Perspectives</strong></p>
                    <p style="margin:3.0pt 0;"><em>Exploring and understanding different virus classification systems and how they impact our responses to disease outbreaks.</em></p>
                    <br>
                    <p><strong>Moderator</strong></p>
                    <ul class="no-bullets"><li>Conrad Schoch, PhD, NCBI</li></ul>
                    <br>
                    <p><strong>Panel and Session Topics</strong> [20-minute talks]</p>
                    <ul class="no-bullets">
                        <li><strong><em>Influenza and SARS-CoV-2: Evolution, Immunity, and Surveillance</em></strong></li>
                        <li>Derek Smith, PhD,  University of Cambridge</li>
                        <li>&nbsp;</li>
                        <li><strong><em>Virus classification and support for different stakeholders</em></strong></li>
                        <li>Peter Simmonds, PhD, University of Oxford</li>
                    </ul>
                    <br>
                    <p><strong>Q&A Panel Discussions</strong> [30 Minutes]</p>
                    <ul>
                        <li>Can virus classification systems be designed to be both clear and adaptable, allowing for consistent communication while effectively tracking virus evolution?</li>
                        <li style="list-style-type: none !important;">
                            <ul>
                                <li>Can these sometimes conflicting goals be effectively addressed, especially during a disease outbreak?</li>
                            </ul>
                        </li>
                        <li>How does rapid virus evolution during outbreaks hinder current sub-species classification methods? Are there alternative approaches better suited for tracking these dynamic changes?</li>
                    </ul>
                    <br>
                    <p>Use Reddit for questions and discussions <a href="https://www.reddit.com/r/BRC_users/comments/1bu5avt/feedback_requested_session_iii_diverse_viewpoints" target="_blank">here</a>.
                </td>
            </tr>
            <tr>
                <td>10:45 AM - 11:00 AM</td>
                <td>Break</td>
            </tr>
            <tr>
                <td>11:00 AM - 12:15 PM</td>
                <td>
                    <p><strong>Session IV: Virus classification schemes</strong></p>
                    <p style="margin:3.0pt 0;"><em>Species and sub-species: Different approaches and schemes for the classification of viruses.</em></p>
                    <br>
                    <p><strong>Moderator</strong></p>
                    <ul class="no-bullets"><li>Richard Scheuermann, PhD, National Library of Medicine (NLM)</li></ul>
                    <br>
                    <p><strong>Panel and Session Topics</strong> [15-minute talks]</p>
                    <ul class="no-bullets">
                        <li><strong><em>The ICTV taxonomy: Classification and nomenclature</em></strong></li>
                        <li>Jens H. Kuhn, PhD, NIH/NIAID/DCR Integrated Research Facility at Fort Detrick</li>
                        <li>&nbsp;</li>
                        <li><strong><em>GISAID classification</em></strong></li>
                        <li>Krista Queen, PhD, LSU Health Shreveport (remote)</li>
                        <li>&nbsp;</li>
                        <li><strong><em>Perspectives on the development of the Pango system and software</em></strong></li>
                        <li>√Åine O'Toole, PhD, The University of Edinburgh</li>
                        <li>&nbsp;</p>
                        <li><strong><em>Pango beyond the SARS-CoV-2 pandemic</em></strong></li>
                        <li>Rachel Colquhoun, PhD, The University of Edinburgh</li>
                    </ul>
                    <br>
                    <p><strong>Q&A Panel Discussion</strong> [15 Minutes]</p>
                    <ul>
                        <li>How do the existing classification schemes impact our ability to track and respond to virus outbreaks?</li>
                        <li>In what ways do these classification schemes influence public health interventions and vaccine development?</li>
                    </ul>
                    <br>
                    <p>Use Reddit for questions and discussions <a href="https://www.reddit.com/r/BRC_users/comments/1bu5ct3/feedback_requested_session_iv_virus" target="_blank">here</a>.
                </td>
            </tr>
            <tr>
                <td>12:15 PM - 1:15 PM</td>
                <td>Lunch</td>
            </tr>
            <tr>
                <td>1:15 PM - 2:30 PM</td>
                <td>
                    <p><strong>Session V: Virus classification tools</strong></p>
                    <p style="margin:3.0pt 0;"><em>Tools to explore virus evolution and their classification.</em></p>
                    <br>
                    <p><strong>Moderator</strong></p>
                    <ul class="no-bullets"><li>Eneida Hatcher, PhD, NCBI</li></ul>
                    <br>
                    <p><strong>Panel and Session Topics</strong> [20-minute talks]</p>
                    <ul class="no-bullets">
                        <li><strong><em>Clade and lineage assignment with Nextclade</em></strong></li>
                        <li>Cornelius Roemer, University of Basel, Switzerland</li>
                        <li>&nbsp;</li>
                        <li><strong><em>UShER and autolin: Identifying virus lineages</em></strong></li>
                        <li>Angie Hinrichs, University of California Santa Cruz</li>
                        <li>&nbsp;</li>
                        <li><strong><em>BV-BRC sub-species classification tools</em></strong></li>
                        <li>Christian Zmasek, J. Craig Venter Institute, BV-BRC</li>
                    </ul>
                    <br>
                    <p><strong>Q&A Panel Discussion</strong> [15 Minutes]</p>
                    <ul>
                        <li>How do different classification systems impact our understanding of virus evolution and disease emergence?</li>
                        <li>Do current classification systems capture sufficient genetic variation and associated phenotypic impact to support prediction of future disease outcomes?</li>
                        <li>Are there emerging classification approaches that hold promise for improved prediction, control and response to virus disease?</li>
                    </ul>
                    <br>
                    <p>Use Reddit for questions and discussions <a href="https://www.reddit.com/r/BRC_users/comments/1bu5ek5/feedback_requested_session_v_virus_classification" target="_blank">here</a>.
                </td>
            </tr>
            <tr>
                <td>2:30 PM - 2:45 PM</td>
                <td>Break</td>
            </tr>
            <tr>
                <td>2:45 PM - 4:00 PM</td>
                <td>
                    <p><strong>Session VI: Analytical and visualization tools</strong></p>
                    <p style="margin:3.0pt 0;"><em>User tools and platforms available to study virus evolution during an outbreak.</em></p>
                    <br>
                    <p><strong>Moderator</strong></p>
                    <ul class="no-bullets"><li>Indresh Singh, J. Craig Venter Institute, BV-BRC</li></ul>
                    <br>
                    <p><strong>Panel and Session Topics</strong> [20-minute talks]</p>
                    <ul class="no-bullets">
                        <li><strong><em>NCBI Virus / Taxonomy</em></strong></li>
                        <li>Eneida Hatcher, PhD, NCBI</li>
                        <li>&nbsp;</li>
                        <li><strong><em>Virus sub-species in ViralZone and UniProtKB</em></strong></li>
                        <li>Philippe Le Mercier, PhD, Swiss-Prot group, Swiss Institute of Bioinformatics, Switzerland</li>
                        <li>&nbsp;</li>
                        <li><strong><em>Thinking about virus mutational spectra through visualization</em></strong></li>
                        <li>Andrew Warren, PhD, University of Virginia, BV-BRC</li>
                    </ul>
                    <br>
                    <p><strong>Q&A Panel Discussion</strong> [15 Minutes]</p>
                    <ul>
                        <li>What role do data repository and analytical platforms provide to support basic research and outbreak response?</li>
                        <li>Are there research and response needs not supported by these platforms?</li>
                        <li>How can these platforms be extended to provide a more  complete spectrum of biomedical data and analytical tools?</li>
                    </ul>
                    <br>
                    <p>Use Reddit for questions and discussions <a href="https://www.reddit.com/r/BRC_users/comments/1bu5gnv/feedback_requested_session_vi_analytical_and" target="_blank">here</a>.
                </td>
            </tr>
            <tr>
                <td>4:00 PM - 4:15 PM</td>
                <td>Break</td>
            </tr>
            <tr>
                <td>4:15 PM - 5:15 PM</td>
                <td colspan="3">
                    <p><strong>White Board Brainstorming Session</strong> [60 minutes]</p>
                    <br>
                    <ul class="no-bullets">
                        <li><strong>Moderators:</strong></li>
                        <li>Wiriya Rutvisuttinunt, PhD, NIAID</p>
                        <li>Session Moderators</li>
                    </ul>
                    <br>
                    <p><strong>Questions:</strong></p>
                    <ul>
                        <li>What are the limitations of current classification systems?</li>
                        <li style="list-style-type: none !important;">
                            <ul>
                                <li>Are they able to track rapidly evolving viruses?</li>
                                <li>Are they able to track associated phenotypic changes?</li>
                                <li>Can they be used to effectively model future genotypic and phenotypic changes are their impact on disease?</li>
                            </ul>
                        </li>
                        <li>Can we use an existing classification approach for all viruses?</li>
                        <li style="list-style-type: none !important;">
                            <ul>
                                <li>If not, how many unique approaches need to be provided/supported?</li>
                                <li>What new tools need to be developed to support classification of all human disease-causing viruses?</li>
                                <li>To what extent can classification be automated and support large volumes of data?</li>
                            </ul>
                        </li>
                        <li>Do we need a standardized nomenclature for evolving lineages?</li>
                        <li style="list-style-type: none !important;">
                            <ul>
                                <li>How do we implement a standardized scheme?</li>
                            </ul>
                        </li>
                        <li>In what ways do these classification schemes influence the public health response?</li>
                    </ul>
                </td>
            </tr>
        </tbody>
    </table>

    <p style="margin-bottom:3.0pt;"><strong>April 10, Day 3: Responding to Public Health Needs</strong></p>
    <table class="docutils align-default workshop-agenda">
        <colgroup>
            <col style="width: 23.0%">
            <col style="width: 77.0%">
        </colgroup>
        <tbody>
            <tr>
                <td>9:00 AM - 9:05 AM</td>
                <td><strong>Welcome back and Recap of Day 1 & 2</strong></td>
            </tr>
            <tr>
                <td>9:05 AM - 10:40 AM</td>
                <td>
                    <p><strong>Session VII: The Public Health Impact, Challenges and the Path Forward</strong></p>
                    <p style="margin:3.0pt 0;"><em>The role of virus classification to support the  public health response to disease and how that classifications helps us identify new threats, track outbreaks, and develop targeted interventions.</em></p>
                    <br>
                    <p><strong>Moderator</strong></p>
                    <ul class="no-bullets"><li>Paul Eder, PhD, NIAID</li></ul>
                    <br>
                    <p><strong>Panel and Session Topics</strong> [20-minute talks]</p>
                    <ul class="no-bullets">
                        <li><strong><em>NIAID: To better understand, treat, and ultimately prevent infectious diseases</em></strong></li>
                        <li>Liliana Brown, Ph.D. NIAID</li>
                        <li>&nbsp;</li>
                        <li><strong><em>CDC: Public Health Perspective: Federal Level</em></strong></li>
                        <li>Andrew Beck, PhD, U.S. Centers for Disease Control and Prevention</li>
                        <li>&nbsp;</li>
                        <li><strong><em>WHO Global Program</em></strong> (remote)</li>
                        <li>Rosamund Lewis, PhD, WHO</li>
                        <li>&nbsp;</li>
                        <li><strong><em>Boots on the ground: State public health laboratory‚Äôs perspective on virus pathogen surveillance</em></strong></li>
                        <li>Sean Wang, Minnesota Department of Public Health</li>
                    </ul>
                    <br>
                    <p><strong>Q&A Panel Discussions</strong> [15 Minutes]</p>
                    <ul>
                        <li>How does an  understanding of virus evolution during outbreaks improve and/or complicate outbreak preparedness and response?</li>
                        <li>How can we improve our understanding of virus evolution to inform the development of more effective vaccines and treatments?</li>
                        <li>Are there any ethical considerations related to the classification and naming of evolving virus lineages with respect to the public health response and the development of interventions?</li>
                    </ul>
                    <br>
                    <p>Use Reddit for questions and discussions <a href="https://www.reddit.com/r/BRC_users/comments/1bu5im7/feedback_requested_session_vii_the_public_health" target="_blank">here</a>.
                </td>
            </tr>
            <tr>
                <td>10:40 AM - 10:55 AM</td>
                <td>Break</td>
            </tr>
            <tr>
                <td>10:55 AM - 12:00 PM</td>
                <td colspan="3">
                    <p><strong>White Board Brainstorming Session</strong> [65 minutes]</p>
                    <br>
                    <ul class="no-bullets">
                        <li><strong>Moderator:</strong></li>
                        <li>Alison St John, PhD, CDC</p>
                    </ul>
                    <br>
                    <p><strong>Panel discussion</strong></p>
                    <ul>
                        <li>What are the needs of the Public Health community?</li>
                        <li style="list-style-type: none !important;">
                            <ul>
                                <li>How do we best meet those needs</li>
                            </ul>
                        </li>
                        <li>Who should develop, implement,  and maintain the classification system(s)?</li>
                        <li style="list-style-type: none !important;">
                            <ul>
                                <li>Will one approach/system suffice, or do multiple systems need to be developed and supported?</li>
                                <li>What other responsibilities need to be supported (e.g., making the tools available along with outreach efforts to publicize the system and train people in their use.)?</li>
                            </ul>
                        </li>
                        <li>Preparing for the next pandemic</li>
                        <li style="list-style-type: none !important;">
                            <ul>
                                <li>How do we bring all of this together in a timely manner?</li>
                                <li>How do we test the proposed response?</li>
                                <li>How will this effort be supported?</li>
                            </ul>
                        </li>
                    </ul>
                    <br>
                    <p><strong>Meeting outcomes and next steps</strong></p>
                    <br>
                    <p><strong>Conclusion</strong></p>
                </td>
            </tr>
        </tbody>
    </table>


.. _speakers-section:
------------------
Workshop Speakers amd Reading List
------------------

---------------------

.. list-table::
   :widths: 20 60 20
   :header-rows: 1

   * - Name
     - Affiliation
     - Papers
   * - **General**
     - Bioinformatics Bottleneck
     - PMID: 33649511
   * - **Anderson, Tavis**
     - U.S. Department of Agriculture
     - PMID: 37208300
   * - **Bahl, Justin**
     - University of Georgia, U.S.
     - 
   * - **Beck, Andrew**
     - U.S. Centers for Disease Control and Prevention (CDC)
     - PMID: 37516478
   * - **Brown, Liliana**
     - NIH/NIAID
     - 
   * - **Colquhoun, Rachel**
     - University of Edinburgh, U.K.
     - PMID: 32669681
   * - **Donato, Celeste**
     - Murdoch Children's Research Institute, Australia
     - PMID: 37323680
   * - **Hatcher, Eneida**
     - National Center for Biotechnology Information (NCBI), U.S.
     - 
   * - **Hinrichs, Angie**
     - University of California Santa Cruz, U.S.
     -  PMID: 38316930, PMID: 38155331.
   * - **Korber, Bette**
     - Los Alamos National Labs, U.S.
     - 
   * - **Kuhn, Jens H.**
     - NIH/NIAID/DCR Integrated Research Facility at Fort Detrick, U.S.
     - PMID: 23001720,  PMID: 37141106
   * - **Le Mercier, Philippe**
     - Swiss-Prot group, Swiss Institute of Bioinformatics, Switzerland
     -  PMID: 37897348
   * - **Lefkowitz, Elliot**
     - University of Alabama at Birmingham (UAB), U.S.
     - PMID: 37710003
   * - **Lewis, Rosamund**
     - World Health Organization
     - PMID: 37349031
   * - **Neher, Richard**
     - University of Basel, Switzerland
     - 
   * - **O'Toole, √Åine**
     - University of Edinburgh, U.K.
     - PMID: 35998195
   * - **Queen, Krista**
     - LSU Health Shreveport, U.S.
     - 
   * - **Roemer, Cornelius**
     - University of Basel, Switzerland
     - joss 03773
   * - **Seto, Donald**
     - George Mason University, U.S.
     - PMID: 38275969, PMID: 34618969
   * - **Simmonds, Peter**
     - University of Oxford, U.K.
     - PMID: 36780432
   * - **Smith, Derek J**
     - University of Cambridge, U.K.
     -  PMID: 15218094
   * - **Wang, Sean**
     - Minnesota Department of Health, U.S.
     - 
   * - **Warren, Andrew**
     - University of Virginia, U.S.
     - PMID: 34520557, PMID: 38361813
   * - **Zmasek, Christian**
     - J.Craig Venter Institute, U.S.
     - PMID: 27981236, PMID: 21034504, PMID: 33972780
Full details of publications will be added soon. 
