# Proteome Comparison Service

The BV-BRC’s Proteome Comparison service can be used to readily identify insertions and deletions in up to nine target genomes that are compared with one reference, which can be a researcher’s private genome in BV-BRC, a genome that has been annotated outside, any of the publicly available genomes in BV-BRC, or a set of proteins that have been saved as a Feature Group in BV-BRC.

The Proteome Comparison service is based on the original Sequence-based Comparison tool that was part of RAST[1]. This tool colors each gene based on protein similarity using BLASTP[2] and marks each gene as either unique, a unidirectional best hit or a bidirectional best hit when compared to the reference genome. The output includes a whole-genome schematic that is colored based on BLAST. A table that details all the results can be downloaded for further analysis, as can a scalable vector graphic (svg) diagram of the results that is publication quality.



![Figure 1](./images/Picture1.png "Figure 1")


![Figure 2](./images/Picture2.png "Figure 2")


![Figure 3](./images/Picture3.png "Figure 3")


![Figure 4](./images/Picture4.png "Figure 4")


![Figure 5](./images/Picture5.png "Figure 5")


![Figure 6](./images/Picture6.png "Figure 6")


![Figure 7](./images/Picture7.png "Figure 7")


![Figure 8](./images/Picture8.png "Figure 8")


![Figure 9](./images/Picture9.png "Figure 9")


![Figure 10](./images/Picture10.png "Figure 10")


![Figure 11](./images/Picture11.png "Figure 11")


![Figure 12](./images/Picture12.png "Figure 12")


![Figure 13](./images/Picture13.png "Figure 13")


![Figure 14](./images/Picture14.png "Figure 14")


![Figure 15](./images/Picture15.png "Figure 15")


![Figure 16](./images/Picture16.png "Figure 16")


![Figure 17](./images/Picture17.png "Figure 17")


![Figure 18](./images/Picture18.png "Figure 18")


![Figure 19](./images/Picture19.png "Figure 19")


![Figure 20](./images/Picture20.png "Figure 20")


![Figure 21](./images/Picture21.png "Figure 21")


![Figure 22](./images/Picture22.png "Figure 22")


![Figure 23](./images/Picture23.png "Figure 23")


![Figure 24](./images/Picture24.png "Figure 24")


![Figure 25](./images/Picture25.png "Figure 25")


![Figure 26](./images/Picture26.png "Figure 26")


![Figure 27](./images/Picture27.png "Figure 27")


![Figure 28](./images/Picture28.png "Figure 28")


![Figure 29](./images/Picture29.png "Figure 29")


![Figure 30](./images/Picture30.png "Figure 30")


![Figure 31](./images/Picture31.png "Figure 31")


![Figure 32](./images/Picture32.png "Figure 32")


![Figure 33](./images/Picture33.png "Figure 33")


![Figure 34](./images/Picture34.png "Figure 34")


![Figure 35](./images/Picture35.png "Figure 35")


![Figure 36](./images/Picture36.png "Figure 36")


![Figure 37](./images/Picture37.png "Figure 37")


![Figure 38](./images/Picture38.png "Figure 38")


![Figure 39](./images/Picture39.png "Figure 39")


![Figure 40](./images/Picture40.png "Figure 40")


![Figure 41](./images/Picture41.png "Figure 41")


