# Comparative Pathway Heatmap

## Accessing the Comparative Pathway Heatmap on the PATRIC Website

This tool is accessible via the “Heatmap” Tab, which is accessed by
clicking on any Pathway Name in the initial search results page
generated within our Comparative pathway Tool. For information on how to
use the Comparative pathway Tool please see [Comparative Pathway Tool
FAQs](/content/Comparative_Pathway).

## Comparative Pathway Heatmap Overview

The Heatmap is an interactive visualization tool, which provides an
overview of the distribution of genomes across the set of EC numbers
within a selected pathway. Patterns visible in a Heatmap can allow for
many types of analysis such as, a bird’s-eye view of the conservation
(or lack thereof) of particular genes of interest, discerning how many
proteins have the same function within a single genome, and identifying
proteins with multiple homologs or paralogs across a set of genomes.

## Color Coding 

Each cell is colored according to how many proteins (features) from a
specific genome are assigned a particular EC number. Black cells
indicate that there are no representative proteins from a specific
genome assigned that EC number. Bright yellow cells indicate there is
one representative protein, dark yellow cells indicate there are two
representative proteins, and dark orange cells indicate there are three
or more representative proteins.

## Data Sorting

-   The entire visualization may be filtered by annotation source. To do
    this, chose an annotation source from the dropdown menu at the top
    of the Heatmap. Note: If you changed the annotation source from the
    default source, PATRIC, on the KEGG Map tab, then your new selection
    will also be the default when you click on the Heatmap tab.

-   Each individual column or row within the Heatmap may be clicked and
    dragged to any specified position.

## Comparative Pathway Heatmap Features and Functionality

-   EC Numbers and Descriptions are listed along the Y (vertical) axis
    and corresponding Genomes are listed along the X (horizontal) axis.
    The scale of these rows and columns may be controlled by sliding the
    x and/or y slidebars located at the axis intersection in the upper
    left corner of the Heatmap.

-   The following options may be accessed from a pop-up widow by
    clicking on any individual colored cell, any EC label, or any Genome
    label within the Heatmap. The same information can be collected for
    a whole region of interest within the Heatmap by clicking and
    dragging to select a group of cells.
    -   The “Download Heatmap Data” button will show the count of
        representative features from each genome assigned to each EC
        Number in either a text or excel file. Note: Clicking this
        button from a EC label will produce a file with the count for
        each genome assigned to that EC number, while clicking on this
        button from a Genome label will produce a file with the count
        for that particular genome in every EC number associated with
        your selected pathway.
    -   The “Add Proteins to Group” button will save selected items to
        new, or existing, groups within your Workspace. To learn more
        about how to save data for future visits and utilize your
        personal Workspace please see [Workspace
        FAQs](/content/Workspace_and_Groups).
    -   Data for each protein (feature), such as Genome Name, Accession
        Number, Locus Tag, Start and End designations, and Overall
        Length may be obtained in several ways:
        -   Clicking the “Download Proteins” button will open the
            protein data in either a text or excel file.
        -   Clicking the “Show Proteins” button will open the protein
            data within a Pathways Table, where the Toolbar, located in
            the light blue row of the table may be utilized to:
            -   Save selected items to groups within your Workspace by
                clicking the Add Feature(s) button. To learn more about
                how to save data for future visits and utilize your
                personal Workspace please see [Workspace
                FAQs](/content/Workspace_and_Groups).
            -   View or download selected DNA and/or protein sequence
                data in FASTA format. The table itself, or selected data
                within it, are also downloadable in both excel and txt
                file formats.
            -   View pathways and associated pathway information for
                your selected genomes by clicking the Pathway Summary
                button.
            -   Conduct Multiple Sequence Alignment analysis on selected
                features by clicking the Multiple Sequence Alignment
                button. For more details on this tool please see
                [Multiple Sequence Alignment
                FAQs](/content/Multiple_Sequence_Alignment).
            -   Utilize the ID Mapping Tool on selected features by
                clicking the Map IDs button. For more information on
                this tool please see [ID Mapping
                FAQs.](/content/ID_Mapping_Tool)

Note: All tables accessed via the Comparative Pathway Heatmap may be
sorted as described in [Feature Table FAQs](/content/Feature_Table).
