BV-BRC (PATRIC & IRD/ViPR) Workshop at the University of Virginia, August 30 - September 1, 2022
================================================================================================

.. feed-entry::
   :date: 2022-07-14

.. image:: ../images/bv-brc-workshop-uva-2022.gif
  :width: 500
  :alt: BV-BRC Workshop UVA

The BV-BRC (Bacterial and Viral Bioinformatics Resource Center) team will be offering a 3-day **Bioinformatics Workshop on August 30 - September 1, 2022, at the University of Virgnia (UVA)** in Charlottesville, VA. The workshop will be held at the UVA North Fork Discovery Park, approximately 10 miles north of UVA grounds.  The workshop will show researchers how to use the new BV-BRC website, which is a merger of the two long running bacterial and viral BRC resources, PATRIC and IRD/ViPR, respectively. The workshop will consist of interactive hands-on training sessions. During the first two days, participants will learn how to search for public datasets of interest and perform genomic, comparative genomic, metagenomic, and transcriptomic analysis using various analysis services and tools at BV-BRC. The third day of the workshop will focus on using command-line interface for programmatic search and retrieval of the data and job submissions, as well as users working with their own data and analysis problems with the help from the BV-BRC team members.

**REGISTRATION**

To register for the workshop, send an email to arw3s@virginia.edu. Registration is limited to 50 people, and will be accepted based on the order the emails are received. You should receive a confirmation email within 2 business days. If you don't receive one, you can also reach us at help@bv-brc.org. 

The workshop is free to attend, but participants must arrange and pay for their own travel, meals, and lodging. Light refreshments will be available. There is also a small coffee and snack bar in the building and a cafe and grill in the adjacent building.

.. cut::

**Note:** *The following COVID-19 safety precautions are currently in place at the University of Virginia.*

* Wearing face masks is optional on UVA properties including the Biocomplexity Institute.

* We will keep the seating at least 6 feet apart at the workshop venue to maintain safe physical distance.

* Full information can be found in `UVA Policy SEC-045: COVID-19 Health & Safety Requirements <https://uvapolicy.virginia.edu/policy/SEC-045>`_.

**LOCATION**

| Biocomplexity Institute
| Town Center Four, 1st Floor, Room A
| 994 Research Park Boulevard
| Charlottesville, VA 22911

**DIRECTIONS**

.. raw:: html

    <p><iframe allowfullscreen="" aria-hidden="false" frameborder="0" height="450" src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d3137.600677340614!2d-78.43706844854414!3d38.14947729818487!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x89b47726a5709f6f%3A0x1de8b16581ced920!2s994%20Research%20Park%20Boulevard%2C%20Charlottesville%2C%20VA%2022911!5e0!3m2!1sen!2sus!4v1603911363299!5m2!1sen!2sus" style="border:0;" tabindex="0" width="600"></iframe></p>

There is plenty of free parking adjacent to the Biocomplexity Institute.

The `Charlottesville Albemarle Airport <http://www.gocho.com/>`_ is conveniently located 2 miles south of the Biocomplexity Institute. There are several hotels and dining options nearby.

**AGENDA**

**Day 1 - Tuesday, August 30**
::

   9:00 am   Introduction
              * Welcome 
              * BV-BRC registration
              * Overview (www.bv-brc.org)

  10:00 am   FASTQ Utilities
              * Desription of FASTQ files and service 
              * Selecting pipeline (Trim, FastQC, Paired read, Align)
              * Uploading reads and submitting job
              * Viewing and interpreting results

  10:45 am   Taxonomic Classification
              * Description of Kraken2
              * Uploading reads or contigs and submitting job
              * Saving classified or unclassified sequences
              * Viewing and Interpreting the results

  11:15 am   Break

  11:30 am   Similar Genome Finder Service
              * Description of MASH/MinHash algorithm
              * Uploading genome sequences, reads, or contigs
              * Submitting job
              * Viewing  and interpreting results
  
  12:00 pm   Lunch

   1:00 pm   Comprehensive Genome Analysis Service
              * Description of assembly and annotation algorithms
              * Uploading reads or contigs and submitting job
              * Viewing and interpreting results

   1:45 pm   Metagenomic Binning
              * Description of algorigthm
              * Uploading reads or contigs and submitting the job
              * Viewing and interpreting the results

   2:15 pm   Break

   2:30 pm   Phylogenetic Tree Building Service
              * Creating a genome group
              * Determining if selected genomes are “treeable”
              * Description of algorithm and submitting tree-building job
              * Viewing and interpreting results
              * Newick file download 

   3:15 pm   Comparative Services (Protein Families, Pathways, and Subsystems)
              * Description of BV-BRC protein families and job submission
              * Finding the pan, core and accessory genomes 
              * Visualizing and manipulating the heatmap viewer 
              * Finding specific differences, downloading and saving results
              * Comparing pathways on pathway map and heatmap
              * Finding specific differences, downloading and saving results
              * Comparing subsystems in table and on heatmap
              * Finding specific differences, downloading and saving results

   4:45 pm   Question and Answer Session and Hands-on Work

   5:00 pm   Adjourn

**Day 2 - Wednesday, August 31**
::

   9:00 am   Review of Day 1

   9:15 am   Proteome Comparison
              * Selecting genomes for a study
              * Visualization of compared genomes
              * Download and analysis of results

   10:00 am   Compare Region Viewer
              * Feature Overview
              * Description of calculations for gene neighborhoods
              * Adjusting the view 
              * Seeing data of interest (genome and feature groups) 

  10:30 pm   Break

  10:45 am   Primer Design
              * Description of algorithm
              * Adjusting parameters and job submission
              * Viewing and interpreting the results

  11:30 pm   Genome Alignment
              * Description of MAUVE
              * Selection of genomes and job submission
              * Viewing the results

  12:00 pm   Lunch

   1:00 pm   BLAST
              *	Description of BLAST
              *	Choosing BLAST database, parameters, and job submission
              *	Viewing and interpreting the results

   1:30 pm   SNP and MNP Variation Service
              *	Description of SNP callers and aligners
              *	Uploading reads 
              *	Discussion and selection of target genomes
              *	Viewing and interpreting the results

   2:00 pm   Gene Tree
              *	Description of algorithm
              *	Selection of data and job submission
              *	Viewing and interpresting the results

   2:30 pm   Break

   2:45 pm   RNA-Seq Pipeline
              *	Discussion of algorithms and choosing a strategy
              *	Uploading RNA-seq data
              *	Selecting genomes  and job submission
              *	Viewing and interpreting the results

   4:00 pm   Install Command Line Interface
              * Installing the Command Line Interface

   4:30 pm   Question and Answer Session and Hands-on Work

   5:00 pm   Day 2 Adjourn

**Day 3 - Thursday, September 1**
::

   9:00 am   Command Line Interface
              * Logging in
              * Searching for data
              * Creating groups from data selections
              * Downloading data

  10:30 am   Break

  10:45 am   Job submission via the command line
              * Uploading private data (singular or batch)
              * Submitting assembly jobs (singular or batch)
              * Submitting annotation jobs (singular or batch)

  12:00 pm   Lunch

   1:00 pm   Working on specific use cases and participant data

   4:00 pm   Final questions

   5:00 pm   Workshop concludes