BV-BRC Workshop at Argonne National Laboratory, December 14-16, 2022
=====================================================================

.. feed-entry::
   :date: 2022-10-07

.. image:: ../images/2022/bv-brc-dec-workshop-anl.jpg
   :width: 500
   :alt: BV-BRC December Workshop at Argonne National Laboratory

   .. role:: notice-header
   :class: notice-header

:notice-header:`NOTICE:`

.. role:: notice-body
     :class: notice-body

:notice-body:`Dear BV-BRC users, we have reached the limit for the maximum number of in-person participants for this workshop. The registration is closed. Please subscribe to the BV-BRC mailing list or follow us on social media to learn about upcoming workshops and webinars.`


The BV-BRC (Bacterial and Viral Bioinformatics Resource Center) team will be offering a **Bioinformatics Workshop on December 14-16, 2022, at Argonne National Laboratory** in the suburbs of Chicago, IL. The workshop will show researchers how to analyze and explore **antimicrobial resistance data** using the BV-BRC website, which is a merger of the two long running bacterial and viral BRC resources, PATRIC and IRD/ViPR, respectively.

The workshop will consist of interactive hands-on training sessions. During the first two days, participants will learn how to search for public datasets of interest and perform genomic, comparative genomic, metagenomic, and transcriptomic analysis using various analysis services and tools at BV-BRC. The third day of the workshop will focus on using command-line interface for programmatic search and retrieval of the data and job submissions, as well as users working with their own data and analysis problems with the help from the BV-BRC team members.

**REGISTRATION**

The workshop will be limited to 50 people on a first-come, first-served basis. There is no fee for the workshop, but participants are responsible for their own travel expenses, accommodation, and meals.

If you wish to participate, please fill out the “Argonne Visitor Request” form at: https://apps.anl.gov/registration/visitors as soon as possible — especially non-U.S. Citizens, since they will take additional time to be cleared. Please enter Lisa Hundley, lhundley@anl.gov, for the “Sponsor’s E-mail Address.”

After you have completed registration, please send a mail to Lisa Hundley ( lhundley@anl.gov ) to confirm your registration.

**COVID-19:**
*Please note the COVID-19 restrictions and safety guidance currently in place at the Argonne National Laboratory.*

* All visitors must be fully vaccinated or provide accepted proof of negative COVID-19 test results that are less than 72 hours old.

* Face coverings are required, except to the extent necessary to eat or drink when maintaining appropriate physical distance, or when an individual is isolated in an enclosed space.

* We will keep the desks at least 6 feet apart from each other at the workshop venue to maintain safe physical distance.

* There are outdoor seating areas with gazebos, which can be used during breaks and lunch time. Face coverings may be removed while outside and socially distancing.   


**LOCATION**

| Building 240 Conference Center - Room 1416
| Argonne National Laboratory
| 9700 Cass Avenue
| Lemont, IL 60439

**AGENDA**

Day 1 - Wednesday, December 14
::

   9:00 am   Argonne information, BV-BRC registration, Overview (www.bv-brc.org)           

   10:00 am   FASTQ Utilities
              * Description of FASTQ files and service 
              * Selecting pipeline (Trim, FastQC, Paired read, Align)
              * Uploading reads and submitting job
              * Viewing and interpreting results

  10:45 am   Taxonomic Classification
              * Description of Kraken2
              * Uploading reads or contigs and submitting job
              * Saving classified or unclassified sequences
              * Viewing and interpreting the results

  11:15 am   Break

  11:30 am   Similar Genome Finder Service
              * Description of MASH/MinHash
              * Uploading genome sequences, reads or contigs
              * Submitting job
              * Viewing  and interpreting results

  
  12:00 pm   Lunch

   1:00 pm   Comprehensive Genome Analysis Service
              * Description of assembly and annotation algorithms
              * Uploading reads or contigs and submitting job
              * Viewing and interpreting results

   1:45 pm   Metagenomic Binning
              * Description of algorithm
              * Uploading reads or contigs and submitting the job
              * Viewing and interpreting the results


   2:15 pm   Break

   2:30 pm   Phylogenetic Tree Building Service
              * Creating a genome group
              * Determining if selected genomes are â€œtreeableâ€
              * Description of algorithm and submitting tree-building job
              *	Viewing and interpreting results
              *	Newick file download 


   3:00 pm   Comparative Services (Protein Families, Pathways and Subsystems)
              * Description of protein families and job submission
              * Finding the pan, core, and accessory genomes 
              * Visualizing and manipulating the heatmap viewer 
              * Finding specific differences, downloading, and saving results into private workspace
              * Comparing pathways on pathway map and heatmap
              * Finding specific differences, downloading, and saving results into private workspace
              * Description of Subsystems
              * Exploring subsystems using the tabular and heatmap views
              * Finding specific differences, downloading, and saving results


   4:45 pm   Question and Answer Session and Hands on Work

   5:00 pm   Day 1 Adjourn

Day 2 -Thursday, December 15
::

   9:00 am   Review of Day 1

   9:15 am   Proteome Comparison
          * Feature Overview
          * Description of calculations for gene neighborhoods
          * Adjusting the view
          * Seeing data of interest (genome and feature groups)

   10:00 am   Compare Region Viewer
           * Genome selection and job selection
           * Comparing pathways on pathway map and heatmap
           * Finding specific differences, downloading, and saving results into private workspace

  10:30 am   Break

  10:45 am   Primer Design
            * Description of algorithm
            * Adjusting parameters and job submission
            * Viewing and interpreting the results


  11:30 am   Genome Alignment 
           * Description of MAUVE
           * Selection of genomes and job submission
           * Viewing the results

  12:00 pm   Lunch

   1:00 pm   BLAST
              *	Description of BLAST
              *	Choosing BLAST database, parameters, and job submission
              *	Viewing and interpreting the results

   1:30 pm   SNP and MNP Variation Service
              *	Description of SNP callers and aligners
              *	Uploading reads 
              *	Discussion and selection of target genomes
              *	Viewing and interpreting the results

   2:00 pm   Gene Tree
              *	Description of algorithm
              *	Selection of data and job submission
              *	Viewing and interpreting the results

   2:30 pm   Break


   2:45 pm   Multiple Sequence Alignment and SNP view 
             * Description of algorithm
             * Selection of genes and job submission
             * Viewing and interpreting the results

   3:00 pm   RNA-Seq Pipeline
              *	Discussion of algorithm and choosing a strategy
              *	Uploading RNA-seq data
              *	Selecting genomes  and job submission
              *	Viewing and interpreting the results


   4:00 pm   Install Command line interface
             * Installing the Command Line Interface

   4:15 pm   Question and Answer Session and Hands on Work

   5:00 pm   Day 2 Adjourn

Day 3 - Friday, December 16
::

   9:00 am	   Command Line Interface 
                * Logging in
                * Searching for data
                * Creating groups from data selections
                * Downloading data

   10:30 am	   Break

   10:45 am	   Job submission via command line 
                * Uploading private data (singular or batch)
                * Submitting assembly jobs (singular or batch)
                * Submitting annotation jobs (singular or batch)
                * Discussion of command line submission to other services 

   12:00 pm    Lunch

   1:00 pm     Working on specific use cases and participant data

   4:00 pm     Final questions

   5:00 pm     Workshop concludes

